bootstrap/controllers/ck8sconfig_controller.go

```diff
@@ -391,6 +397,39 @@ func (r *CK8sConfigReconciler) resolveFiles(ctx context.Context, cfg *bootstrapv
 	return collected, nil
 }
 
+func (r *CK8sConfigReconciler) resolveInPlaceUpgradeRelease(machine *clusterv1.Machine) cloudinit.SnapInstallData {
+	mAnnotations := machine.GetAnnotations()
+
+	if mAnnotations != nil {
+		val, ok := mAnnotations[bootstrapv1.InPlaceUpgradeReleaseAnnotation]
```

Comment:
You could invert the if statement here to remove a level of indentation, check if mAnnotations == nil and early return.